### DISCUSSION TITLE
Could not locate Comsol (Windows)

### DISCUSSION THREAD

**Discussion by bierchen65** [MPh-py/MPh#150] [General]
Hi
Although COMSOL is installed in the default path, I get the error message "_C_ould not locate any Comsol_ installation._ . 
My COMSOL.exe is located at C:/Program Files/COMSOL/COMSOL54/Multiphysics/bin/win64

I use **Windows** 10 so Mph should find the location.
I could not solve the problem because I do not know the destination path for a symbolic link on windows.

**Comment by john-hen** [comment]
But the [Installation] chapter also says "Versions 5.5 and newer are expected to work." While you have Comsol 5.4, which is thus *not* expected to work.

[Installation]: https://mph.readthedocs.io/en/1.2/installation.html

**Reply by bierchen65** [reply]
Thanks for the quick reply

**Reply by john-hen** [reply]
If you are in any mood to debug this — because you can't, for license reasons, upgrade to a newer Comsol version — let me know. I, for one, do note have access to a Comsol 5.4 installation on Windows. But if you feel like hacking the code, there certainly is a way to get this to work. Not a way that will make it into library code, as I cannot test any of that. But I can assist in debugging.

MPh does not "simply" have to find the Comsol installation folder. It also needs to know where the Java run-time environment is located, within the Comsol folder. And it needs to find all the Comsol Java plug-ins, somewhere inside the Comsol folder, that provide the actual "Comsol" functionality. This hasn't changed "in essence" between Comsol versions, but clearly has changed "somewhat". Which is why  5.4 doesn't work. I have no insight into why it doesn't, but I might be able to help.
